src/quest_manager/views.py

```diff
@@ -381,6 +381,70 @@ def post(self, request, *args, **kwargs):
         return redirect("quests:archived")
 
 
+class QuestBulkEditView(UserPassesTestMixin, View):
+    # Note: TAs pass this permission check, but bulk editing UI is currently restricted to staff only,
+    # so TAs cannot perform bulk edits via the frontend yet.
+    def test_func(self):
+        return is_staff_or_TA(self.request.user)
+
+    def post(self, request, *args, **kwargs):
+        """
+        Handle bulk editing operations on selected quests, including:
+        - Deletion
+        - Publishing
+        - Unpublishing
+        - Unarchiving
+
+        Expects:
+            - 'selected_quests[]': A list of quest IDs to act upon.
+            - 'action': The bulk action to perform (e.g., "delete", "publish").
+
+        Redirects to the quest list view with a success or warning message.
+
+        Returns:
+            HttpResponseRedirect: Redirect to the quest list page with feedback.
+        """
+        quest_ids = request.POST.getlist("selected_quests[]")
+        action = request.POST.get("action")
+
+        if not quest_ids or not action:
+            messages.warning(request, "No quests selected.")
+            return redirect("quests:quests")
+        if action in ["unarchive", "delete"]:
+            # Include archived quests for delete and unarchive actions because
+            # those target quests excluded from the default queryset.
+            quests = Quest.objects.all_including_archived().filter(id__in=quest_ids)
+        else:
+            quests = Quest.objects.filter(id__in=quest_ids)
+
+        if action == "delete":
+            count = quests.count()
+            quests.delete()
```

Comment:
Does this bulk action trigger signals (map update signals and user quest available list signals etc)?